### PR TITLE
feat: derive debug clone traits on some items

### LINKED ISSUE #2395: derive copy on DeclareResult
### Which component is your feature related to?

Forge

### Feature Request

```rust
#[derive(Drop, Serde, Clone)]
enum DeclareResult {
    Success: ContractClass,
    AlreadyDeclared: ContractClass,
}
```

-> 
```rust
#[derive(Drop, Serde, Copy)]
enum DeclareResult {
    Success: ContractClass,
    AlreadyDeclared: ContractClass,
}
```

There's no reason to be deriving Clone, when ContractClass derives Copy.

### PR DESCRIPTION
**Pull request by enitrat**
Closes #2395

## Introduced changes

- derives Debug and Clone on types for easier external use and debug
- made some items public (no effect)

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -28,26 +34,29 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Bumped Cairo version to `2.7.0`
 - Max steps in tests (configured by argument `--max-n-steps`) now defaults to 10 million
-if not provided (changed from 4 million).
+  if not provided (changed from 4 million).
```
Comment (Arcticae) [inline CHANGELOG.md:37]: Can you revert those (unwanted) formatting changes? I'd rather modify this file incrementally

**Review by Arcticae — APPROVED** [review]
LGTM, one nitpick

**Comment by enitrat** [comment]
> Server returned 404 Not Found for https://book.cairo-lang.org/ch14-01-contract-storage.html#addresses-of-storage-variables

I refactored the entire storage section of the book so this fails in CI

**Comment by Arcticae** [comment]
@enitrat https://github.com/foundry-rs/starknet-foundry/pull/2440 this one removes the link

**Pull request merged**